components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -608,6 +608,8 @@ private AccessTokenDO createNewTokenBean(OAuthTokenReqMessageContext tokReqMsgCt
         newTokenBean.setTokenBinding(tokReqMsgCtx.getTokenBinding());
         newTokenBean.setAccessTokenExtendedAttributes(
                 getAccessTokenExtendedAttributes(tokenReq.getAccessTokenExtendedAttributes(), tokReqMsgCtx));
+        newTokenBean.setAcr(tokReqMsgCtx.getSelectedAcr());
```

Comment:
We don't need these values for all the access tokens. Can you add these conditionally only when requested.